Robot.java_03_14_17

```diff
@@ -1,418 +1 @@
-	 * the robot is disabled.
-	 */
-	@Override
-	public void disabledInit() {
-		this.left.disable();
-		this.right.disable();
-
-	}
-
-	@Override
-	public void disabledPeriodic() {
-		Scheduler.getInstance().run();
-	}
-
-	/**
-	 * This autonomous (along with the chooser code above) shows how to select
-	 * between different autonomous modes using the dashboard. The sendable
-	 * chooser code works with the Java SmartDashboard. If you prefer the
-	 * LabVIEW Dashboard, remove all of the chooser code and uncomment the
-	 * getString code to get the auto name from the text box below the Gyro
-	 *
-	 * You can add additional auto modes by adding additional commands to the
-	 * chooser code above (like the commented example) or additional comparisons
-	 * to the switch structure below with additional strings & commands.
-	 */
-	@Override
-	public void autonomousInit() {
-		autonomousCommand = chooser.getSelected();
-		
-		
-		double KpR=9/72.0;
-		double KpL=9/72.0;
-		double Ki = 0.01/72.0;
-		double Kd= 1/72.0;
-		//double KfR= 1.4/36.0;
-		//double KfL=1.45/36.0;
-		double Kf = 1.0;
-		
-		this.right = new Good_DriveSide(KpR, Ki, Kd, Kf, rightEncoder, rightSpark);
-		right.setInverted(true);
-		this.left = new Good_DriveSide(KpL, Ki, Kd, Kf, leftEncoder, leftSpark);
-
-		
-		driveTrain = new RobotDrive(left,right);
-
-		/*
-		 * String autoSelected = SmartDashboard.getString("Auto Selector",
-		 * "Default"); switch(autoSelected) { case "My Auto": autonomousCommand
-		 * = new MyAutoCommand(); break; case "Default Auto": default:
-		 * autonomousCommand = new ExampleCommand(); break; }
-		 */
-		//autonomousCommand = new MoveForward();
-		// schedule the autonomous command (example)
-		if (autonomousCommand != null)
-			autonomousCommand.start();
-		
-
-		myTimer.reset();
-		myTimer.start();
-		
-		
-		System.out.println("Autonomous!");
-		System.out.println("StartTime: " + myTimer.get());
-		System.out.println();
-		System.out.println();
-		
-	//leftEncoder.reset();
-		gyro.calibrate();
-//		System.out.println("gyro calibration: " + gyro.getAngle());
-		
-		
-		
-	}
-
-	/**
-	 * This function is called periodically during autonomous
-	 */
-	@Override
-	public void autonomousPeriodic() {
-		
-		Scheduler.getInstance().run();
-	
-		//	System.out.println("encoder count: "+ ((Good_DriveSide) left).getEncoderCount());
-			//QUESTION: if i made my own method inside good_driveside, which implements speedController
-			//do I have to cast left as a goodDriveside instead of speedcontroller?
-		
-		if (state.equals("forward1")) {
-			moveForward(9.5167);
-			//10.83 feet = 129.96 inches
-			//9.5167 feet = 114.3 inches
-			//have to calculate again with the length of the robot and the distance from the center
-			//from left and right positions not the middle
-			state = "turn1";
-				
-		}
-		else if (state.equals("turn1")) {
-			changeAngle(120); 
-			state = "forward2";
-			
-		}
-		else if (state.equals("forward2")) {
-			moveForwardSlow(.5);
```

Comment:
The way this is coded now, both changeAngle and moveForwardSlow read the absolute value from the sensor. So, this moveForwardSlow will never actually do anything because when it is called, the robot has already moved 9.5, which is greater that 0.5.

Also, related to another comment I made, you want to think about these functions as being called every run loop (and do not have a while loop inside). So, this means that setting the state right after calling the function, will cause the states to cycle through very quickly, before the robot is done what you want. A good way to deal with this isssue is to define the functions moveForward, changeAngle etc, to return True if the move is complete and False otherwise. Then, in this loop, you can check if the step is done. If it is, you change the state, and reset the count on the sensors to zero to make sure that in the next step, you are starting from zero.

To reset the encoders to zero, the function is encoder.reset(). http://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/Encoder.html